substrate/primitives/core/src/crypto.rs

```diff
@@ -419,6 +419,16 @@ pub fn set_default_ss58_version(new_default: Ss58AddressFormat) {
 	DEFAULT_VERSION.store(new_default.into(), core::sync::atomic::Ordering::Relaxed);
 }
 
+/// Interprets the string `s` in order to generate a public key without password.
+#[cfg(feature = "serde")]
```

Comment:
> I am now thinking It is a bit confusing to mandate using `serde` feature to have access to the method. Not very clear why they would be related. Can we not feature guard `alloc::format` behind `serde`, so that we don't feature guard `get_public_from_string_or_panic` either?

I also don't know why feature guard `alloc::format` behind `serde`, but I worry it causes breaking if not feature. Or we can import `format` inside `get_public_from_string_or_panic` if don't use `serde`

Reply:
We can pull alloc::format w/o any feature gates (just like vec).

FYI: there is a script:
https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/core/check-features-variants.sh
which checks all needed features combinations.

Reply:
> We can import `alloc::format` w/o any feature gates (just like vec::Vec).
> 
> FYI: there is a script: https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/core/check-features-variants.sh which checks all needed features combinations.

Ok